app/controllers/jira/status/push_controller.rb

```diff
@@ -35,13 +35,16 @@ def edit; end
 
       def deploy_email
         flash_text =
-          if @push.email_sent?
+          if @push.email_sent? || !@push.update_attributes(email_sent: true)
```

Comment:
This isn't quite true that a false return from `update_attributes` would mean the email is already sent (it could be some [unrelated] validation error on the model in question)

I would feel better if we skipped validations and callbacks (unless the `email_sent` change has any callbacks you are relying on?) and use the low level update_column method:

```
if @push.email_sent?
  'Email was already sent.'
else
  begin
    @push.update_column(:email_sent, true)
    DeployMailer.deployment_email(@push.jira_issues).deliver_now!
    'Email has been sent.'
  rescue
    @push.update_column(:email_sent, false)
    raise
  end
end
```